agents/postgres/parser/parser_test.go

```diff
@@ -0,0 +1,390 @@
+package parser
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestExtractTables(t *testing.T) {
+	tests := []struct {
+		name    string
+		query   string
+		want    []string
+		wantErr bool
+	}{
+		{
+			"simple select query",
```

Comment:
I think it would be better to move those queries to `testdata` files:
* `queryXXX.sql` for queries, this way they can be executed by `psql` directly;
* `queryXXX.json` with expected results (table names or error).

As a bonus point, the same data can be used in benchmarks and fuzzing.